src/sunmd5_fmt_plug.c

```diff
@@ -553,319 +539,314 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 		MD5_Final(data[idx].digest, &data[idx].context);
 	}
 
-	/*
-	 * now to delay high-speed md5 implementations that have stuff
-	 * like code inlining, loops unrolled and table lookup
-	 */
-
-	/* this is the 'first' sprintf(roundascii,"%d",round);  The rest are at the bottom of the loop */
-	strcpy(roundascii, "0");
-	roundasciilen=1;
-
-	for (round = 0; round < maxrounds; round++) {
+#ifdef _OPENMP
+#pragma omp parallel for default(none) private(idx) copyin(input_buf_big) \
+	shared(saved_salt, data, constant_phrase, ngroups, group_sz)
```

Comment:
Got a problem here. GCC won't allow me to mark `constant_phrase` as shared, as it's already a constant variable. But icc wants to keep it anyway. Now the situation is: keeping it, gcc fails to compile; dropping it, icc fails to compile. Damn.